flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,185 @@
+            raise DependencyError(
+                plugin_name, 'paho.mqtt.client', 'paho-mqtt python module is required for MQTT notify plugin. ImportError: %s' % e
+            )
+
+        def on_log_cb(client, userdata, level, buff):
+            logger.verbose(str(buff))
+
+        def on_publish_cb(client, userdata, mid):
+            logger.verbose('MQTT on_publish callback -  message was successfully published to broker as messageID={}',str(mid))
+            client.disconnect()
+
+        def on_disconnect_cb(client, userdata, rc):
+            logger.verbose('MQTT on_disconnect callback - disconnected with result code {} [{}]',str(rc),conn_rc_description_map.get(rc),'Unknown')
+            client.loop_stop()
+            
+        config['title'] = title
+        config['message'] = message
+        config['payload'] = '{} - {}'.format(config.get('title'),config.get('message'))
+
+        conn_rc_description_map = { 0: 	'Connection Accepted',
+                            1: 	'Connection Refused, unacceptable protocol version - The Server does not support the level of the MQTT protocol requested by the Client',
+                            2: 	'Connection Refused, identifier rejected - The Client identifier is correct UTF-8 but not allowed by the Server',
+                            3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                            4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                            5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+        MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
+                           'MQTTv311': 	mqtt.MQTTv31 }
+
+        logger.trace('MQTT notify config={}',str(config))
+
+        client = mqtt.Client(protocol=MQTT_proto_map.get(config.get('broker_protocol',mqtt.MQTTv311)), transport=config.get('broker_transport', 'tcp') )
+
+        client.enable_logger(logger=logger)
+
+        client.on_log = on_log_cb
+        client.on_publish = on_publish_cb
+        client.on_disconnect = on_disconnect_cb
+
+        #Handle SSL/TLS communication w/out certificate authentication
+        if not config.get('certificates',{}).get('client_cert',False) and config.get('enable_encrypted_communication',False):
+            client.tls_set(ca_certs=certs.get('broker_ca_cert'),
+                          certfile=None,
+                          keyfile=None,
+                          cert_reqs=ssl.CERT_NONE)
+
+            client.tls_insecure_set(True)
+
+            logger.verbose('Basic SSL/TLS encrypted communications enabled')
+            logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+
+        #Handle SSL/TLS communication with certificate authentication
+        if config.get('certificates',False):
+            certs = config.get('certificates',{})
```

Comment:
since you already know that `config` hold `certificates`, you could afford to be a little less defensive here:
```suggestion
            certs = config['certificates']
```

Reply:
I was just keeping consistent on accessing all config settings using the ```.get()``` methods instead of mix-n-match.  I can go either way if you really prefer this way.

Reply:
best practice is to access member directly if you assume its there (or you want an exception to be raised if it isn't)

Reply:
addressed with commit 09fba9c